src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -99,12 +103,66 @@ protected String name() {
             return "Property Mutator";
         }
 
-        private final Caller mMutator;
-        private final Caller mAccessor;
+        private final PropertyCaller mMutator;
+        private final PropertyCaller mAccessor;
         private final WeakHashMap<View, Object> mOriginalValues;
         private final Object[] mOriginalValueHolder;
     }
 
+    public static class LayoutSetVisitor extends ViewVisitor {
+        public LayoutSetVisitor(List<Pathfinder.PathElement> path, JSONObject args) {
+            super(path);
+            mOriginalValues = new WeakHashMap<View, JSONObject>();
+            mArgs = args;
+        }
+
+        @Override
+        public void cleanup() {
+            for (Map.Entry<View, JSONObject> original:mOriginalValues.entrySet()) {
+                final View changedView = original.getKey();
+                final JSONObject originalValue = original.getValue();
+                setLayout(changedView, originalValue.optInt("verb"), originalValue.optInt("anchor"));
+            }
+        }
+
+        @Override
+        public void accumulate(View found) {
+            final int newRuleIndex = mArgs.optInt("verb");
```

Comment:
It isn't obvious, but, in EditProtocol, I did
```
params.put("verb", layout_info.getInt("rule_id"));
```
so the verb will exist for sure. Otherwise, the code won't reach this point.